lib/jsonapi/plugs/content_type_negotiation.ex

```diff
@@ -49,11 +47,32 @@ defmodule JSONAPI.ContentTypeNegotiation do
     {conn, content_type}
   end
 
-  defp respond({conn, @jsonapi, @jsonapi}) do
+  defp respond({conn, nil, nil}), do: add_header_to_resp(conn)
+  defp respond({conn, @jsonapi, nil}), do: add_header_to_resp(conn)
+  defp respond({conn, nil, @jsonapi}), do: add_header_to_resp(conn)
+  defp respond({conn, @jsonapi, @jsonapi}), do: add_header_to_resp(conn)
+  defp respond({conn, content_type, nil}) do
+    case validate_header(content_type) do
+      true -> add_header_to_resp(conn)
+      false -> send_error(conn, 415)
+    end
+  end
+  defp respond({conn, nil, accepts}) do
+    case validate_header(accepts) do
+      true -> add_header_to_resp(conn)
+      false -> send_error(conn, 406)
+    end
+  end
+  defp respond({conn, @jsonapi, _accepts}), do: send_error(conn, 406)
+  defp respond({conn, _content_type, _accepts}), do: send_error(conn, 415)
```

Comment:
not sure if this is getting out of hand or just how it is supposed to be :man_shrugging:

Reply:
I think I need to add another case where both content-type and accept header are passed and need to parse through each one, throwing 415 or 406, whichever one hits first.

Reply:
which can then replace the last two pattern matched functions